### PR TITLE
dev:Full data and MC samples for 2022 and 2023

### PR DESCRIPTION
**Pull request by uttiyasarkar**
The most updated samples are added for Run3 2022 and 2023 data and MC. The list below are the selected NanoAOD samples.

Open tasks:

- [x] update QCDEMenriched cross-sections
- [x] update drell-yan cross-sections
- [x] possible indentation problem 

Screenshot 2024-08-07 at 21 02 34
Screenshot 2024-08-07 at 21 44 33

### CODE REVIEW

**Comment by andreypz** [comment]
Thanks for starting this dev. I have a few suggestions so far:
- [x] The `sample` field should be our nickname of the sample, not the dataset name. Check out the 2017 config for the names used there and use the same. These needs to be edited manually.
- [x] The output json file should not be different for each datasets. This is also our choice. The convention I used for other configs is: `Run3_DATA`, 'Run3_MC_Sig', `Run3_MC_VJets`, `Run3_MC_OtherBkg`. Could you use the same convention.
- [x] Not sure if it is necessary to split 2022 and 2023. Is there a reason for this? If not I suggest to combine them. I did split the 2016,2017,2018 before. But I think this was not necessary. We could have listed them in a single file. 
- [x] The Pt-Njet binned samples for Z+jets are still missing, and they are most important to add.
- [x] ZNuNu and WJets samples needed (inclusive and binned) 
- [x] Hcc signlal samples are of course needed too.

**Comment by uttiyasarkar** [comment]
All of the above mentioned missing samples are added now in one combine samples_Run3.json replacing the old json.

The only missing item:

- [x] Cross-section of multiple processes are still missing

File: samples_Run3.json [hunk 1]
```diff
@@ -1,380 +1,4682 @@
...
+        ],
+        "metadata": {
+          "year": "2023_postBPix",
+          "isMC": true,
+          "xsec": "6346.0"
+        }
+      },
+      {
+        "das_names": [
+          "/DYto2L-2Jets_MLL-50_TuneCP5_13p6TeV_amcatnloFXFX-pythia8/Run3Summer22NanoAODv12-130X_mcRun3_2022_realistic_v5_ext1-v1/NANOAODSIM"
```
Comment (andreypz) [inline samples_Run3.json]: This is an extension sample (`ext1` in the name). This is fine, but then should also add a nominal sample as well, I think

File: samples_Run3.json [hunk 2]
```diff
@@ -1,380 +1,4682 @@
...
+        ],
+        "metadata": {
+          "year": "2023_postBPix",
+          "isMC": true,
+          "xsec": "96.9"
+        }
+      },
+      {
+        "das_names": [
+          "/TTto2L2Nu_TuneCP5_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv12-130X_mcRun3_2022_realistic_v5_ext1-v2/NANOAODSIM"
```
Comment (andreypz) [inline samples_Run3.json]: This is an extension sample (`ext1` in the name), but missing the nominal. Should we also add the nominal sample in the list?

File: samples_Run3.json [hunk 3]
```diff
@@ -1,380 +1,4682 @@
...
+        ],
+        "metadata": {
+          "year": "2022_preEE",
+          "isMC": true,
+          "xsec": "96.9"
+        }
+      },
+      {
+        "das_names": [
+          "/TTto2L2Nu_TuneCP5_13p6TeV_powheg-pythia8/Run3Summer22EENanoAODv12-130X_mcRun3_2022_realistic_postEE_v6_ext1-v2/NANOAODSIM"
```
Comment (andreypz) [inline samples_Run3.json]: Also an extension

File: samples_Run3.json [hunk 4]
```diff
@@ -1,380 +1,4682 @@
...
+        ],
+        "metadata": {
+          "year": "2022_postEE",
+          "isMC": true,
+          "xsec": "19317.5"
+        }
+      }
+    ]
+  },
+  "DYJetsToLL_MLL-50_FxFx": {
```
Comment (andreypz) [inline samples_Run3.json]: Could you change the name to `DYJetsToLL_FxFx` to be consistent with previous definition. The MLL 50 cut is always assumed in this sample

File: samples_Run3.json [hunk 5]
```diff
@@ -1,380 +1,4682 @@
...
+        "metadata": {
+          "year": "2022_postEE",
+          "isMC": true,
+          "xsec": null
+        }
+      }
+    ]
+  },
+  "DYJetsToLL_MLL-50_MLM": {
+    "sample": "DYto2L_MLL-50_MLM",
```
Comment (andreypz) [inline samples_Run3.json]: Could you change the name to `DYJetsToLL_MLM`

File: samples_Run3.json [hunk 6]
```diff
@@ -1,380 +1,4682 @@
...
+        "metadata": {
+          "year": "2022_preEE",
+          "isMC": false,
+          "primaryDataset": "SingleMuon",
+          "era": "A"
+        }
+      },
+      {
+        "das_names": [
+          "/SingleMuon/Run2022A-22Sep2023-v1/NANOAOD"
```
Comment (andreypz) [inline samples_Run3.json]: Added twice?

**Review by andreypz — CHANGES_REQUESTED** [review]
Thanks for the updates, but I have more comments/requests.
Some of them are inline, others here:

* You have added Zto2Q and Wto2Q samples (HT binned), but I don't think these are needed. I know I also had one of the bins in 2017, but this was for a test. I think the contribution from them would be minimal. But don't remove them, just note that these are not crucial.
*  Instead we need the Zto2L and WtoLNu Pt-binned FxFx (NLO) samples. You have added Zto2Nu, which is great. The similar should exist for Zto2L and WtoLNu.
* In some cases you used extensions of the samples, without adding the nominal sample. Maybe you could also add the nominal in the list. 
* Could you please double check the differences in your changed compared to the previous version. Are there any differences and why? 
* Also, I did not see the WH_Hto2C signal samples. Could you add those.

**Comment by andreypz** [comment]
And one more thing: could you use 4 spaces for indents instead of 2 spaces. I think this is more conventional and would be consistent with other PocketCoffea analyses configs

File: samples_Run3.json (hunk 5 above)
Comment (uttiyasarkar) [reply samples_Run3.json]: Added different Mass binned samples in the newest push such as MLL-4to50, MLL-50to120, MLL-10to50, MLL-50. Do we keep only one type? Then MLL-50 name is redundant.

File: samples_Run3.json (hunk 4 above)
Comment (uttiyasarkar) [reply samples_Run3.json]: Added different Mass binned samples in the newest push such as MLL-4to50, MLL-50to120, MLL-10to50, MLL-50. Do we keep only one type? Then MLL-50 name is redundant.

Comment (andreypz) [reply samples_Run3.json]: I don't think we need the mass binned samples. We select Z peak region in the analysis. 
We need the Pt-binned and N-jet binned samples however.

Comment (andreypz) [reply samples_Run3.json]: The samples like this: DYto2L-2Jets_MLL-50_PTLL-100to200_1J_TuneCP5_13p6TeV_amcatnloFXFX-pythia8

File: samples_Run3.json (hunk 2 above)
Comment (uttiyasarkar) [reply samples_Run3.json]: Fixed

**Comment by uttiyasarkar** [comment]
The following samples only exists as ```ext``` samples:

```/DYto2L-2Jets_MLL-10to50_TuneCP5_13p6TeV_amcatnloFXFX-pythia8/Run3Summer23BPixNanoAODv12-130X_mcRun3_2023_realistic_postBPix_v2_ext1-v3/NANOAODSIM```
```/DYto2L-2Jets_MLL-10to50_TuneCP5_13p6TeV_amcatnloFXFX-pythia8/Run3Summer23NanoAODv12-130X_mcRun3_2023_realistic_v14_ext1-v2/NANOAODSIM```

**Comment by uttiyasarkar** [comment]
The PR should be ready for review and merging

**Pull request merged**